RogueSharp/Algorithms/Graph.cs

```diff
@@ -68,15 +68,18 @@ public IEnumerable<int> Adjacent( int vertex )
       public override string ToString()
       {
          var formattedString = new StringBuilder();
-         formattedString.AppendFormat( "{0} vertices, {1} edges {2}", NumberOfVertices, NumberOfEdges, Environment.NewLine );
+         formattedString.AppendFormat( "{0} vertices, {1} edges{2}", NumberOfVertices, NumberOfEdges, Environment.NewLine );
          for ( int v = 0; v < NumberOfVertices; v++ )
          {
-            formattedString.AppendFormat( "{0}: ", v );
+            formattedString.AppendFormat( "{0}:", v );
             foreach ( int vertex in _adjacent[v] )
             {
-               formattedString.AppendFormat( "{0} ", vertex );
+               formattedString.AppendFormat( " {0}", vertex );
+            }
+            if ( v < NumberOfVertices - 1 )
+            {
+               formattedString.AppendLine();
             }
-            formattedString.AppendLine();
          }
          return formattedString.ToString();
       }
```

Comment:
Graph and EdgeWeighedDigraph are largely the same. Maybe they should inherit/extend a base class.